gammapy/estimators/energydependence.py

```diff
@@ -0,0 +1,259 @@
+import numpy as np
+from astropy import units as u
+from gammapy.datasets import Datasets, MapDataset
+from gammapy.modeling import Fit
+from gammapy.modeling.models import FoVBackgroundModel
+from gammapy.stats.utils import ts_to_sigma
+from .core import Estimator
+
+
+def weighted_chi2_parameter_results(table_edep, parameter="sigma"):
+    """Calculate the weighted chi2 value for the parameter of interest"""
+
+    values = table_edep[f"{parameter}"][1:]
+    errors = table_edep[f"{parameter}_err"][1:]
+
+    weights = 1 / errors**2
+    avg = np.average(values, weights=weights)
+
+    chi2_value = np.sum((values - avg) ** 2 / errors**2)
+    df = len(values) - 1
+    sigma_value = ts_to_sigma(chi2_value, df)
+
+    chi2_result = {}
+    chi2_result[f"chi2 {parameter}"] = [chi2_value]
+    chi2_result["df"] = [df]
+    chi2_result["significance"] = [sigma_value]
+
+    return chi2_result
+
+
+class EnergyDependenceEstimator(Estimator):
+    """Test if there is any energy-dependent morphology in a map dataset for a given set of energy bins.
+
+    Parameters
+    ----------
+    energy_edges : `~astropy.units.Quantity`
+        Energy edges for the energy-dependence test.
+    model : `~gammapy.modeling.model.SkyModel`
+        Source model kernel.
+    selection_optional : list of str
+        Which additional quantities to estimate. Available options are:
+            * "src-sig": the significance above the background is
+            calculated in each energy band
+
+    Examples
+    --------
+
+    """
+
+    tag = "EnergyDependenceEstimator"
+    _available_selection_optional = ["src-sig"]
+
+    def __init__(self, energy_edges, model, fit=None, selection_optional=None):
+
+        self.energy_edges = energy_edges
+        self.model = model
+        self.num_energy_bands = len(self.energy_edges) - 1
+
+        if fit is None:
+            fit = Fit(optimize_opts={"print_level": 1})
+
+        self.fit = fit
+        self.selection_optional = selection_optional
+
+    def estimate_source_significance(self, dataset):
```

Comment:
Yes! I actually have another version of the code in which `estimate_source_significance` uses the `TestStatisticNested`, but we decided against it because it requires some `n_sigma` to be specified

Reply:
I think it could be set to `-np.inf` in that case